rfc/halyard-lite.md

```diff
@@ -0,0 +1,475 @@
+these are effectively not secrets once Halyard has written the unencrypted value to
+disk, users can store the unencrypted values directly in the Halyard config until
+deck supports decryption of encrypted secrets.
+* Halyard will no longer decrypt secrets for the purpose of validating the config;
+a reduction of the scope of Halyard's validation is discussed in more detail in
+the previous section.
+
+#### Config backup/restore
+
+Halyard has functionality to backup the entire configuration to an archive file,
+and to restore from such an archive file.  This is useful either for backup
+purposes, or to copy configuration from one Halyard installation to another.
+
+In general, the canonical solution for backing up and tracking changes to configuration
+in a Kubernetes-native world is to store this configuration in a git repository.
+To avoid duplicating existing solutions, Halyard will no longer have built-in
+support for backing up and restoring the configuration; it will instead presume
+that the operator is using some external tool to version and back up the configuration.
+
+#### Version publishing/deprecating
+
+Halyard handles publishing and deprecating Spinnaker versions; this functionality
+is only used by maintainers of the project and requires having a GCP service
+account with write permissions to the GCS bucket that stores Spinnaker version
+info. To simplify the surface area for end users, this functionality will be
+removed from Halyard and migrated to a separate admin-specific tool to handle
+these changes.
+
+## Retained core competency
+
+Having outlined the features to be removed from Halyard above, this section will
+outline the remaining core competency of Halyard.
+
+Halyard will focus on its core competency of translating a Halyard config file
+into the config files for the individual microservices.  Halyard will expose a
+single command that takes as input a Halyard config and outputs to a specified
+directory the configurations for each microservice, as well as files referenced
+by these config files.
+
+This command will roughly map onto the current `hal config generate` command.
+
+### API contract
+
+#### Input
+
+The input to Halyard Lite will be a Halyard config file, expressed as YAML.
+In order to promote backwards compatibility, the format of this config file will
+not change.  There will be some fields that are no longer relevant in Halyard Lite;
+these fields will be documented as being ignored by Halyard Lite and it will emit
+a non-fatal warning if these fields are set.  (An example of such a field is
+`deploymentEnvironment.tolerations`, which does not affect the generation of
+service configurations but only adds fields to the output deployment YAML.)
+
+As the Halyard config is not currently well-documented (with the documentation
+instead living on the hal command reference), we will add extensive documentation
+on this config file.
```

Comment:
Would this documentation live within the code, or be separate? 

I have a strongly held opinion that it should live with the code and create doc artifacts during build. This is similar in direction to an upcoming backend config proposal I have coming through the pipes.

Reply:
This would absolutely live alongside the code; the documentation will effectively just be documentation on the types that comprise the hal config.  I'm currently looking into the best way to auto-generate this among the various tools out there.

Reply:
Added this explicitly to the RFC.